### ISSUE TITLE
Dependency conflicts v0.6.0

### ISSUE THREAD

**Issue by MikeWallis42** [astronomer/astronomer-cosmos#237]
I'm currently struggling with installing cosmos because of project dependencies.
We use the constraints file published by Airflow to ensure that we don't have any clashes.

When trying to install Cosmos 0.6.0 with https://raw.githubusercontent.com/apache/airflow/constraints-2.5.1/constraints-3.10.txt
I get an issue with `astronomer-cosmos 0.6.0 depends on apache-airflow-providers-docker>=3.5.0` the constraints file pins this package to `apache-airflow-providers-docker==3.4.0`
The docker provider gets bumped to 3.5.1 in Airflow 2.5.2 which for me then sets this as the minimum Airflow version.
If we want to support back to Airflow 2.4.0 then is it possible to set the provider version to `apache-airflow-providers-docker>=3.1.0` or is there new functionality that we need?

OK that's fine for us as we're going to be running Airflow 2.5.3 anyway. But it might not be for others.

My next challenge then is with filelock.
In the Airflow constraints file for 2.5.3 it's pinned to `filelock==3.10.7`, virtualenv allows a range of between 3.4.1 and 4.0.0
https://github.com/pypa/virtualenv/blob/20.21.0/pyproject.toml#L40
and cosmos doesn't specify a version, so I think pip is resolving that to the latest version for me.
```
#0 3.371 ERROR: Cannot install filelock==3.11.0 because these package versions have conflicting dependencies.
#0 3.372 
#0 3.372 The conflict is caused by:
#0 3.372     The user requested filelock==3.11.0
#0 3.372     The user requested (constraint) filelock==3.10.7
```
Regardless any use of filelock in the project was removed with this PR #233 so can we please remove it as a project dependency?

**Comment by CorsettiS** [comment]
this is one of the reasons I did not like that much when I saw that cosmos introduced a Docker & k8s option. A previous lightweight package is now dependent on airflow-docker & airflow-kubernetes. 
As far as I remember from the code, both docker & k8s dbt methods are only subclassing the main methods from docker & k8s, therefore loosing up the constraint would definitely work from my perspective.

About filelock, you are right that after #233 it just became a left over in the code.

**Comment by jlaneve** [comment]
Yeah, it looks like we missed this in the review - I don't like having them as direct dependencies. Let me see if I can move them into the optional dependencies, like everything else

**Comment by jlaneve** [comment]
@MikeWallis42 @CorsettiS just opened this PR to fix the issue! https://github.com/astronomer/astronomer-cosmos/pull/238

**Comment by jlaneve** [comment]
this should be fixed with 0.6.1, but let me know if you still see an issue!
